src/modules/crm/lib/import-invoice-addresses.js

```diff
@@ -0,0 +1,93 @@
+const { groupBy, mapValues, flatMap, last, get } = require('lodash');
+const moment = require('moment');
+const { pool } = require('../../../lib/connectors/db');
+const queries = require('./queries');
+const { logger } = require('../../../logger');
+
+const DATE_FORMAT = 'YYYY-MM-DD';
+
+const groupByIASNumber = data => groupBy(data, row => row.IAS_CUST_REF);
+
+const groupByDate = data => groupBy(data, row => row.start_date);
+
+/**
+ * Maps the address changes into the minimum possible set of rows
+ * Where an address is the same for adjacent rows, they are merged.
+ * End date on preceding row set to 1 day before start date of current row
+ * @param  {Array} rows - row data from DB query
+ * @return {Array}      - de-duplicated/mapped data for insertion to CRM
+ */
+const deduplicateAddressChanges = rows => {
+  const grouped = mapValues(groupByIASNumber(rows), rows => {
+    const dateGroups = groupByDate(rows);
```

Comment:
I think this would be better as 

`const dateGroups = groupBy(data, row => row.start_date);`

so it is obvious what is happening.